src/Statement.cpp

```diff
@@ -302,6 +316,11 @@ Statement::make_random(CGContext &cg_context,
 		break;
 	}
 
+	if(!label_attr_generate){
```

Comment:
I'm wondering whether we can find a better place to initialize the `label_attr_generator`. What about in constructor of `StatementFilter`? It can be a singleton instance if I'm correct.

Reply:
> I'm wondering whether we can find a better place to initialize the `label_attr_generator`. What about in constructor of `StatementFilter`? It can be a singleton instance if I'm correct.

The constructor of `StatementFilter` is being called more than once. 

Should I move below code to `StatementFilter` constructor?
`if(!label_attr_generate){
                InitializeLabelAttributes();
                label_attr_generate = true;
        }
`

Reply:
Yes please, make it a static variable and initialize it just once.

Reply:
Does it look fine now?